### PR TITLE
Register language/currency dropdown components

### PR DESCRIPTION
**Pull request by jshimko**
- Refactor and register language/currency dropdown components
- delete old components that got incorrectly added back during the marketplace merge resolution
- remove all usage of the “pure” HOC from recompose for now (to address any re-render issues with i18n and currency)

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -159,7 +159,7 @@
           ],
           "alias": {
             "@reactioncommerce/reaction-collections": "./imports/plugins/core/collections",
-            "@reactioncommerce/reaction-components": "/imports/plugins/core/components/lib",
+            "@reactioncommerce/reaction-components": "./imports/plugins/core/components/lib",
```
Comment (aaronjudd) [inline package.json]: I had been wondering why there was a discrepancy here...

**Comment by jshimko** [comment]
We really need to get this reviewed and merged soon.  Aside from the fact that it may fix a few things with i18n and currency, it's also going to end up being a daily merge conflict because of how many files were touched.

File: imports/plugins/core/i18n/client/components/languageDropdown.js [hunk 2]
```diff
@@ -1,6 +1,6 @@
 import React, { Component } from "react";
 import PropTypes from "prop-types";
-import { Button, Divider, DropDownMenu, MenuItem } from "/imports/plugins/core/ui/client/components";
+import { Components } from "@reactioncommerce/reaction-components";
```
Comment (aaronjudd) [inline imports/plugins/core/i18n/client/components/languageDropdown.js]: Typo?
`Import { Components } from  " @actioncommerce / reaction-components " ;`

File: imports/plugins/core/i18n/client/containers/currencyDropdown.js [hunk 3]
```diff
@@ -1,18 +1,13 @@
-import React, { Component } from "react";
+import { compose, withProps } from "recompose";
 import { Meteor } from "meteor/meteor";
 import { Match } from "meteor/check";
 import { Reaction } from "/client/api";
-import { composeWithTracker } from "@reactioncommerce/reaction-components";
+import { registerComponent, composeWithTracker } from "@reactioncommerce/reaction-components";
```
Comment (aaronjudd) [inline imports/plugins/core/i18n/client/containers/currencyDropdown.js]: Typo

File: imports/plugins/core/i18n/client/containers/localizationSettings.js [hunk 4]
```diff
@@ -1,61 +1,64 @@
...
-  }
+    handleEnableAllLanguages = (isEnabled) => {
+      Meteor.call("shop/updateLanguageConfiguration", "all", isEnabled);
+    }
 
-  handleEnableAllCurrencies = (isEnabled) => {
-    Meteor.call("shop/updateCurrencyConfiguration", "all", isEnabled);
-  }
+    handleEnableAllCurrencies = (isEnabled) => {
+      Meteor.call("shop/updateCurrencyConfiguration", "all", isEnabled);
```
Comment (aaronjudd) [inline imports/plugins/core/i18n/client/containers/localizationSettings.js]: What's with the spacing?

**Comment by aaronjudd** [comment]
Are those code weirdness I saw just some GitHub artifacts?

File: imports/plugins/core/i18n/client/components/languageDropdown.js (hunk 2 above)
Comment (jshimko) [inline imports/plugins/core/i18n/client/components/languageDropdown.js]: Are we looking at the same thing?  I don't see that.

register_language_currency_dropdown_components_by_jshimko_ _pull_request__2670_ _reactioncommerce_reaction

File: imports/plugins/core/i18n/client/containers/currencyDropdown.js (hunk 3 above)
Comment (jshimko) [inline imports/plugins/core/i18n/client/containers/currencyDropdown.js]: Same here.  I don't see any typos.

register_language_currency_dropdown_components_by_jshimko_ _pull_request__2670_ _reactioncommerce_reaction

File: imports/plugins/core/i18n/client/containers/localizationSettings.js (hunk 4 above)
Comment (jshimko) [inline imports/plugins/core/i18n/client/containers/localizationSettings.js]: The whole class got indented because it's no wrapped properly.

**Comment by jshimko** [comment]
Are you viewing it in the Github client locally or something?  I don't see any of these issues.  The build also wouldn't have worked if any of those typos existed.

**Comment by jshimko** [comment]
Thanks!

**Pull request merged**